templates/patroni.yml.j2

```diff
@@ -2,6 +2,12 @@ bootstrap:
   dcs:
     postgresql:
       use_pg_rewind: true
+      remove_data_directory_on_rewind_failure: true
+      remove_data_directory_on_diverged_timelines: true
```

Comment:
Are we sure this is necessary, and won't happen by accident? Surely if rewind fails, we should make it possible for a user to go into the database, fix whatever's blocking rewind, then try again?

Reply:
About `remove_data_directory_on_rewind_failure`, if pg_rewind fail to synchronize the data from the new primary to the former primary for some reason (like pg_rewind not being able to write some file, or other things) it probably will leave the data directory in an unrecoverable state and the recommended way to solve that is to remove the data directory, according the PostgreSQL documentation (warning section from https://www.postgresql.org/docs/current/app-pgrewind.html).

The other setting, `remove_data_directory_on_diverged_timelines`, will delete the data directory only if, for some reason, it is not possible to use pg_rewind. And this process would be started in the same situation as pg_rewind would.

Both settings will not be used to delete the data directory if the connection is not possible to the former primary or if the new primary has not run a checkpoint (written data from the Write Ahead Logs files, to the final data files - something that is important to make it a consistent instance to follow). It would happen only in the situations described above.

Reply:
ok, thanks for the detailed explanation :)